tenant_schemas/tests/test_routes.py

```diff
@@ -25,106 +28,50 @@ def setUp(self):
-        """
-        Request path should not be altered.
-        """
-        with self.settings(DEFAULT_SCHEMA_NAME=self.non_exisitant_tenant.schema_name):
-            self.tm = DefaultSchemaTenantMiddleware()
-
-            request_url = '/any/request/'
-            request = self.factory.get('/any/request/',
-                                       HTTP_HOST=self.tenant_domain)
-            self.tm.process_request(request)
-
-            self.assertEquals(request.path_info, request_url)
-
-            # request.tenant should also have been set
-            self.assertEquals(request.tenant, self.tenant)
+    def test_non_existent_tenant_routing(self):
+        """Raise 404 for unrecognised hostnames."""
+        request = self.factory.get(
+            self.url, HTTP_HOST=self.non_existent_tenant.domain_url)
+        self.assertRaises(Http404, self.tm.process_request, request)
 
     def test_non_existent_tenant_to_default_schema_routing(self):
-        """
-        Request path should not be altered.
-        """
-        with self.settings(DEFAULT_SCHEMA_NAME=self.tenant.schema_name):
-            self.tm = DefaultSchemaTenantMiddleware()
-
-            request_url = '/any/request/'
-            request = self.factory.get('/any/request/',
-                                       HTTP_HOST=self.non_exisitant_domain)
-            self.tm.process_request(request)
-
-            self.assertEquals(request.path_info, request_url)
-
-            # request.tenant should also have been set
-            self.assertEquals(request.tenant, self.tenant)
-
-    def test_non_existent_tenant_to_non_existing_default_schema_routing(self):
-        """
-        Request path should not be altered.
-        """
-        with self.settings(DEFAULT_SCHEMA_NAME=self.non_exisitant_tenant.schema_name):
-            self.tm = DefaultSchemaTenantMiddleware()
-
-            request = self.factory.get('/any/request/',
-                                       HTTP_HOST=self.non_exisitant_domain)
-
-            self.assertRaises(self.tm.TENANT_NOT_FOUND_EXCEPTION,
-                              self.tm.process_request, request)
-
-    def test_non_existent_tenant_default_schema_not_set_routing(self):
-        """
-        Request path should not be altered.
-        """
-        self.tm = DefaultSchemaTenantMiddleware()
-
-        request = self.factory.get('/any/request/',
-                                   HTTP_HOST=self.non_exisitant_domain)
+        """Route unrecognised hostnames to the 'public' tenant."""
+        request = self.factory.get(
+            self.url, HTTP_HOST=self.non_existent_tenant.domain_url)
+        self.dtm.process_request(request)
+        self.assertEquals(request.path_info, self.url)
+        self.assertEquals(request.tenant, self.public_tenant)
 
-        self.assertRaises(self.tm.TENANT_NOT_FOUND_EXCEPTION,
-                          self.tm.process_request, request)
+    def test_non_existent_tenant_custom_middleware(self):
```

Comment:
I will add a test case that illustrates my reasons for using SuspiciousTenantMiddleware to cover this.

On my phone at present, will do it when on a decent device :)